### PR TITLE
no word wrap markdown in comments

### PR DESCRIPTION
**Pull request by romainfrancois**
### Intent

addresses (part of) #8378 

### Approach

no word wrap when entering markdown in comments

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by romainfrancois** [comment]
This is not perfect I guess, but I think this improves the sitation:

- entering or exiting markdown with ` ``` ` will disable/enable wrapping. 
- when we see a bullet, wrapping is temporarily disabled for this line
- trying to deal with Rd macros as well: 
  - entering a macro will up a macro depth counter and disable wrapping
  - exiting macro (a line that just has `}` ) will down the counter and enable the wrapping if counter back to 0. 

https://user-images.githubusercontent.com/2625526/191745096-5eb7aa3b-5419-4f11-afbc-cf5b0d1b5752.mov

**Comment by romainfrancois** [comment]
I'll deal with the tab stop thing from #8378 separately.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 1]
```diff
@@ -5093,8 +5095,38 @@ protected void onChunkWritten(String chunk,
          else if (content.trim().startsWith("@"))
             wordWrap.setWrappingEnabled(true);
 
+         if (content.matches("^\\s*```"))
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:5098]: Do we need to (or does this already) support things like:

    ``` r
    ```r

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 2]
```diff
@@ -5093,8 +5095,38 @@ protected void onChunkWritten(String chunk,
...
+            wordWrap.setWrappingEnabled(false);
+
+         boolean macroStart = content.matches("^\\s*\\\\[a-zA-Z]+\\{.*");
+         if (macroStart) 
+         {
+            macroDepth = macroDepth + 1;
+            wordWrap.setWrappingEnabled(false);
+         }
+
+         boolean macroEnd = content.matches("^\\s*\\}.*");
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:5116]: Do we need to handle things like `\preformatted{hello}` here? Ie, things where the closing `}` isn't on its own line?

It also might be easier to choose a solution that just counts matching `{` and `}` pairs, and just keep a stack of different "states" associated with the "type" of opening `{`.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 3]
```diff
@@ -5093,8 +5095,38 @@ protected void onChunkWritten(String chunk,
...
+         if (macroEnd) 
+         {
+            macroDepth = macroDepth - 1;
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:5119]: We should guard against unmatched `}` here (make sure `macroDepth` is always >= 0)

**Comment by romainfrancois** [comment]
I think this is improved, but probably needs a lot more testing

**Comment by romainfrancois** [comment]
The multi paragraph indent makes it difficult

**Comment by romainfrancois** [comment]
I'm not so sure about the macros, i.e. the reflow does not interfere and just let them as is. But it's not a big deal because they're not that used anymore, and this is an improvement over the previous behaviour. 

Same for markdown chunks, they just stay as is.

**Review by kevinushey — APPROVED** [review]
LGTM!

**Pull request merged**